### PR TITLE
Add Cloudbackup Delete API to delete backups

### PR DESCRIPTION
**Pull request by disrani-px**
The previous Delete has been renamed to DeleteAll to delete all backups for a
given volume

Signed-off-by: Dinesh Israni <disrani@portworx.com>

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -296,6 +296,15 @@ type CloudBackupEnumerateResponse struct {
 }
 
 type CloudBackupDeleteRequest struct {
```
Comment (pault84) [inline api/api.go]: Someone will need to document these API's (and i'm not suggesting myself :) )

File: api/server/volume.go [hunk 2]
```diff
@@ -951,6 +951,7 @@ func (vd *volAPI) Routes() []*Route {
 		{verb: "POST", path: backupPath("/restore", volume.APIVersion), fn: vd.cloudBackupRestore},
 		{verb: "GET", path: backupPath("", volume.APIVersion), fn: vd.cloudBackupEnumerate},
 		{verb: "DELETE", path: backupPath("", volume.APIVersion), fn: vd.cloudBackupDelete},
+		{verb: "DELETE", path: backupPath("/all", volume.APIVersion), fn: vd.cloudBackupDeleteAll},
```
Comment (pault84) [inline api/server/volume.go]: old api should be deprecated if we don't intend on using it anymore. Again this should happen through swagger api and then removed from next release.

We should also release not this as a breaking change. (same with all the other cloudsnap api renames.

File: api/api.go (hunk 1 above)
Comment (disrani-px) [reply api/api.go]: I'll file a separate bug for that. All the cloudbackup APIs need to be documented.

File: api/server/volume.go (hunk 2 above)
Comment (disrani-px) [reply api/server/volume.go]: Old API will still be used.

Changes for undocumented APIs don't need to be release noted as it is expected that they can change. Once we documents them we'll release note breaking changes (we should not be making breaking changes for documented APIs in the first place)

File: api/api.go (hunk 1 above)
Comment (disrani-px) [reply api/api.go]: Issue #362 to track that.

**Pull request merged**